core/arch/arm/sm/sm_a32.S

```diff
@@ -183,6 +183,15 @@ UNWIND(	.cantunwind)
 	cmp	r0, #0
 	beq	.sm_ret_to_nsec
 
+	add	r0, sp, #(SM_CTX_NSEC + SM_NSEC_CTX_R0)
```

Comment:
Please patch the c-code in `sm_from_nsec()` instead.
When entering and exiting to nsec we can avoid saving and restoring some registers.